x/oracle/types/oracle.go

```diff
@@ -13,19 +14,73 @@ func (m SignedOracleRegistrationVote) ValidateBasic() error {
 	if len(m.OracleRegistrationVote.UniqueId) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
 	}
+
 	if len(m.OracleRegistrationVote.UniqueId) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "uniqueID is empty")
 	}
-	_, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VoterAddress)
-	if err != nil {
+
+	if _, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VoterAddress); err != nil {
 		return err
 	}
-	_, err = sdk.AccAddressFromBech32(m.OracleRegistrationVote.VotingTargetAddress)
-	if err != nil {
+
+	if _, err := sdk.AccAddressFromBech32(m.OracleRegistrationVote.VotingTargetAddress); err != nil {
 		return err
 	}
+
 	if len(m.OracleRegistrationVote.EncryptedOraclePrivKey) == 0 {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "encryptedOraclePrivKey is empty")
 	}
 	return nil
 }
+
+func (m Oracle) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(m.Address); err != nil {
+		return sdkerrors.Wrapf(err, "oracle address is invalid. address: %s", m.Address)
+	}
+	return nil
+}
+
+func (m OracleRegistration) ValidateBasic() error {
```

Comment:
Like `Oracle.ValidateBasic()`, some fields are not cared for basic validation. I think the belows should be also handled in this ValidateBasic function. wdyt?

- NodePubKeyRemoteReport : not nil
- VotingPeriod : not nil
- TallyResult : can be nil or valid type of TallyResult

Reply:
Thank you.
What does valid type mean in TallyResult?

Reply:
If the TallyResult is not nil, all the fields in TallyResult should not be nil.

Reply:
The `Yes`, `No`, `InvalidYes` field inside TallyResult is `sdk.Int`.
I can't do a nil check, but I can do a negative check.

This negative check and consensusValue(->byte array) will be checked nil.

Reply:
Thank you.
I fixed it.(https://github.com/medibloc/panacea-core/pull/366/commits/3919243b28c5bdbcd393f86c719662138602f480)